server/datastore/mysql/software_titles.go

```diff
@@ -0,0 +1,198 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/jmoiron/sqlx"
+)
+
+func (ds *Datastore) SoftwareTitleByID(ctx context.Context, id uint) (*fleet.SoftwareTitle, error) {
+	const selectSoftwareTitleStmt = `
+SELECT
+	st.id,
+	st.name,
+	st.source,
+	COUNT(DISTINCT hs.host_id) AS hosts_count,
```

Comment:
I can't use the `software_host_counts` table because a host might have more than one version of the software installed.

I went with this momentarily hoping to validate if we need to store some form of cached aggregate like we do for software versions in the load test, but happy to tweak if anyone already sees a potential issue here.